### PR TITLE
DBMS-Session2-Assignment_1

### PR DESCRIPTION
**Pull request by meta-ramjilal-choudhary**
Only assignment 1 submitted 
remaining 2 assignments will be submitted tomorrow
@metaamitasharma

### CODE REVIEW

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+CREATE DATABASE ShopCart;
+SHOW DATABASES;
+USE ShopCart;
+SHOW tables;
+
+#create CATEGORY table-----------------
+CREATE TABLE category(
+    category_id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    category_title VARCHAR(20) NOT NULL,
+    category_desc VARCHAR(100) NOT NULL,
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: Description is optional generally. Do you have such requirement here ?

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    parent_category_id INT,
+    FOREIGN KEY(parent_category_id) REFERENCES category(category_id)
+);
+
+#create PRODUCT table-------------------
+CREATE TABLE product(
+    product_id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    product_title VARCHAR(50) NOT NULL,
+    product_desc VARCHAR(100) NOT NULL,
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: Again !

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    product_date DATE NOT NULL,
+    product_price DOUBLE NOT NULL,
+    product_quantity INT NOT NULL,
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: Why all these fields required ?

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+CREATE TABLE order_line(
+    order_line_id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    user_order_id INT NOT NULL,
+    product_id INT NOT NULL,
+    product_quantity INT NOT NULL,
+    order_status VARCHAR(30) NOT NULL
+);
+
+#create ORDER table----------------------
+CREATE TABLE user_order(
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: can be order simply.

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    FOREIGN KEY(product_id) REFERENCES product(product_id)
+);
+
+#create order_line table
+CREATE TABLE order_line(
+    order_line_id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    user_order_id INT NOT NULL,
+    product_id INT NOT NULL,
+    product_quantity INT NOT NULL,
+    order_status VARCHAR(30) NOT NULL
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: Order status is not expected to fill on creation of any record. You can give default value like - placed.

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+);
+
+#create ORDER table----------------------
+CREATE TABLE user_order(
+    user_order_id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: Order address ? User can have multiple addresses.

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    user_order_date DATE NOT NULL,
+    user_order_amount DOUBLE NOT NULL,
+    user_id INT NOT NULL,
+    FOREIGN KEY(user_id) REFERENCES user(user_id)
+);
+
+#create PAYMENT table--------------------
+CREATE TABLE payment(
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: Just create a column in user_order i.e. payment_method. No need to create separate table as we discussed.

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    user_order_id INT,
+    FOREIGN KEY(user_order_id) REFERENCES user_order(user_order_id)
+);
+
+#create USER table----------------------
+CREATE TABLE user(
+    user_id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    user_name VARCHAR(50) NOT NULL,
+    user_email VARCHAR(50) NOT NULL,
+    user_mobile CHAR(11) NOT NULL
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: again required ?

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+    address_id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    user_id INT NOT NULL,
+    user_address VARCHAR(100) NOT NULL,
+    FOREIGN KEy(user_id) REFERENCES user(user_id)
+);
+
+#create ROLES table--------------------
+CREATE TABLE roles(
+    roles_id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    roles_type VARCHAR(20) NOT NULL,
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: role_title

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+    user_id INT NOT NULL,
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: I think user will have role_id not the role will have user_id

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+    FOREIGN KEY(user_id) REFERENCES user(user_id)
+);
+
+#Show all tables present in DATABASE
+SHOW TABLES;
+
+#Drop Product table---------------------
+drop table product;
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: Although command line is case insensitive, but follow the case conventions for better practice.

File: DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt (hunk 3 above)
Comment (meta-ramjilal-choudhary) [reply DBMS/DBMS-Session2-Assignment_1/Assignment_1.txt]: product date use for product added on that date
product_price - unit price of a product
product_quantity - total stock quantity
@metaamitasharma

**Pull request closed without merging**